### PR TITLE
Fix WeightAveraging swapping in the un-updated average model during validation

### LINKED ISSUE #21724: EMA validation swap models before first update_parameters
### Bug description

With `EMAWeightAveraging` and a delayed start (e.g. update_starting_at_step: 1000), validation still swaps `pl_module` and `_average_model` on every val epoch (e.g. val every 100 steps), even while `self._average_model.n_averaged == 0`. The `_average_model` is only a deepcopy(pl_module) from setup() and has never been updated. Retrieval metrics before `update_starting_at_step` near 0.

Example for EMA init:
```yaml:
use_buffers: false
decay: 0.999
update_every_n_steps: 1
update_starting_at_step: 1000
update_starting_at_epoch: -1
```

Suggested fix: in `WeightAveraging` [on_validation_epoch_start](https://github.com/Lightning-AI/pytorch-lightning/blob/master/src/lightning/pytorch/callbacks/weight_averaging.py#L216) / [on_validation_epoch_end](https://github.com/Lightning-AI/pytorch-lightning/blob/master/src/lightning/pytorch/callbacks/weight_averaging.py#L230), only call `_swap_models` when `self._average_model.n_averaged > 0`.

### What version are you seeing the problem on?

v2.6

### Reproduced in studio

_No response_

### How to reproduce the bug

```python

```

### Error messages and logs

_No response_

### Environment

_No response_

### More info

_No response_

cc @ethanwharris

### PR DESCRIPTION
**Pull request by ATOM00blue**
## What does this PR do?

Fixes #21724
closes #21754

`WeightAveraging` (and its `EMAWeightAveraging` subclass) creates the `AveragedModel` in `setup()` as a copy of the model's initial weights, with `n_averaged == 0`. The validation hooks `on_validation_epoch_start` / `on_validation_epoch_end` swapped this average model in unconditionally whenever it existed.

When using a delayed start (e.g. `EMAWeightAveraging(update_starting_at_step=1000)`) and validating during the warmup period, the average model has never been updated, so validation ran against the **frozen initial (untrained) weights** instead of the current trained ones. This is what the issue describes as metrics being near zero before `update_starting_at_step`.

This PR only swaps the models for validation once the average model has actually been updated at least once (`n_averaged > 0`). The swap remains balanced across the start/end hooks because `n_averaged` does not change during validation.

### Tests

- Added `test_weight_averaging_no_swap_before_first_update`, which verifies that during a never-reached delayed start the parameters seen at validation are the current trained weights, not the frozen initial snapshot. It fails before this change and passes after.
- Updated `SWATestCallback` swap-count expectations: with a delayed update schedule, validation now only swaps once the average model has been updated.

<details>
  <summary><b>Before submitting</b></summary>

- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you write any **new necessary tests**?
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you **update the CHANGELOG**?

</details>

## PR review

Anyone in the community is welcome to review the PR.

### CODE REVIEW

**Comment by Martovark** [comment]
I think it's better to use `self._latest_update_step > 0` from `WeightAveraging` instead of `self._average_model.n_averaged > 0`, because most likely `self._average_model.n_averaged` lives on cuda and it requires device <-> host sync.

**Comment by deependujha** [comment]
Hi @Martovark, great catch. Verified it from:

https://github.com/pytorch/pytorch/blob/0d62256a2b23365f8e1604297eb23a6545102aa8/torch/optim/swa_utils.py#L283-L285

Fixed standalone tests; should be good to land. Thanks @ATOM00blue for the work.

**Review by deependujha — APPROVED** [review]
